### PR TITLE
Add CI for releases

### PR DESCRIPTION
**Pull request by wzshiming**
/kind feature

~~Blocked by #11~~

``` bash
> git tag -f v0.1.0
> DRY_RUN=true PUSH=true make cross-image

docker buildx build --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/kwok:v0.1.0 --platform=linux/amd64 --platform=linux/arm64 --push -f images/kwok/Dockerfile .

> DRY_RUN=true PUSH=true make cross-build

GOOS=linux GOARCH=amd64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/linux/amd64/kwok ./cmd/kwok
gsutil cp -P ./bin/linux/amd64/kwok gs://k8s-staging-kwok-gcb/v0.1.0/bin/linux/amd64/kwok
GOOS=linux GOARCH=amd64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/linux/amd64/kwokctl ./cmd/kwokctl
gsutil cp -P ./bin/linux/amd64/kwokctl gs://k8s-staging-kwok-gcb/v0.1.0/bin/linux/amd64/kwokctl
GOOS=linux GOARCH=arm64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/linux/arm64/kwok ./cmd/kwok
gsutil cp -P ./bin/linux/arm64/kwok gs://k8s-staging-kwok-gcb/v0.1.0/bin/linux/arm64/kwok
GOOS=linux GOARCH=arm64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/linux/arm64/kwokctl ./cmd/kwokctl
gsutil cp -P ./bin/linux/arm64/kwokctl gs://k8s-staging-kwok-gcb/v0.1.0/bin/linux/arm64/kwokctl
GOOS=darwin GOARCH=amd64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/darwin/amd64/kwok ./cmd/kwok
gsutil cp -P ./bin/darwin/amd64/kwok gs://k8s-staging-kwok-gcb/v0.1.0/bin/darwin/amd64/kwok
GOOS=darwin GOARCH=amd64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/darwin/amd64/kwokctl ./cmd/kwokctl
gsutil cp -P ./bin/darwin/amd64/kwokctl gs://k8s-staging-kwok-gcb/v0.1.0/bin/darwin/amd64/kwokctl
GOOS=darwin GOARCH=arm64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/darwin/arm64/kwok ./cmd/kwok
gsutil cp -P ./bin/darwin/arm64/kwok gs://k8s-staging-kwok-gcb/v0.1.0/bin/darwin/arm64/kwok
GOOS=darwin GOARCH=arm64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/darwin/arm64/kwokctl ./cmd/kwokctl
gsutil cp -P ./bin/darwin/arm64/kwokctl gs://k8s-staging-kwok-gcb/v0.1.0/bin/darwin/arm64/kwokctl
GOOS=windows GOARCH=amd64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/windows/amd64/kwok.exe ./cmd/kwok
gsutil cp -P ./bin/windows/amd64/kwok.exe gs://k8s-staging-kwok-gcb/v0.1.0/bin/windows/amd64/kwok.exe
GOOS=windows GOARCH=amd64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/windows/amd64/kwokctl.exe ./cmd/kwokctl
gsutil cp -P ./bin/windows/amd64/kwokctl.exe gs://k8s-staging-kwok-gcb/v0.1.0/bin/windows/amd64/kwokctl.exe
GOOS=windows GOARCH=arm64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/windows/arm64/kwok.exe ./cmd/kwok
gsutil cp -P ./bin/windows/arm64/kwok.exe gs://k8s-staging-kwok-gcb/v0.1.0/bin/windows/arm64/kwok.exe
GOOS=windows GOARCH=arm64 go build -ldflags '-X sigs.k8s.io/kwok/pkg/consts.Version=v0.1.0' -o ./bin/windows/arm64/kwokctl.exe ./cmd/kwokctl
gsutil cp -P ./bin/windows/arm64/kwokctl.exe gs://k8s-staging-kwok-gcb/v0.1.0/bin/windows/arm64/kwokctl.exe

> DRY_RUN=true PUSH=true make cross-cluster-image

docker buildx build --build-arg=kube_version=v1.24.2 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.24.2 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.23.6 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.23.6 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.22.9 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.22.9 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.21.12 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.21.12 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.20.15 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.20.15 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.19.16 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.19.16 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.18.20 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.18.20 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.17.17 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.17.17 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.16.15 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.16.15 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.15.12 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.15.12 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker buildx build --build-arg=kube_version=v1.14.10 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.14.10 --platform=linux/amd64 --platform=linux/arm64 --push -f images/cluster/Dockerfile .
docker build --build-arg=kube_version=v1.13.12 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.13.12 -f images/cluster/Dockerfile .
docker push registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.13.12
docker build --build-arg=kube_version=v1.12.10 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.12.10 -f images/cluster/Dockerfile .
docker push registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.12.10
docker build --build-arg=kube_version=v1.11.10 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.11.10 -f images/cluster/Dockerfile .
docker push registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.11.10
docker build --build-arg=kube_version=v1.10.13 --build-arg=kwok_version=v0.1.0 --tag=registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.10.13 -f images/cluster/Dockerfile .
docker push registry.k8s.io/kwok/cluster:v0.1.0-k8s.v1.10.13
```

### CODE REVIEW

**Comment by wzshiming** [comment]
# https://github.com/kubernetes/test-infra/tree/master/config/jobs/image-pushing/
# k8s-staging-kwok.yaml
``` yaml
postsubmits:
  # this is the github repo we'll build from; this block needs to be repeated for each repo.
  kubernetes-sigs/kwok:
    - name: post-kwok-push-images
      cluster: k8s-infra-prow-build-trusted
      annotations:
        # this is the name of some testgrid dashboard to report to.
        testgrid-dashboards: sig-scheduling
      decorate: true
      branches:
        - ^main$
        - ^release-.*
        # Build tags also
        # this is a regex for semver, from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
        - ^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
      spec:
        serviceAccountName: gcb-builder
        containers:
          - image: gcr.io/k8s-staging-test-infra/image-builder:v20220428-ae431ed1aa
            command:
              - /run.sh
            args:
              # this is the project GCB will run in, which is the same as the GCR images are pushed to.
              - --project=k8s-staging-kwok
              - --scratch-bucket=gs://k8s-staging-kwok-gcb
              - --env-passthrough=PULL_BASE_REF
              - --with-git-dir
              - .
```

**Comment by wzshiming** [comment]
This CI process has not been tried yet, and I don't know if there is access to push images and binaries, so maybe we can release a v0.0.1-alpha.0 to test it first.

/cc @Huang-Wei @yibozhuang

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,20 @@ BUCKET ?= gs://k8s-staging-kwok-gcb
 
 GIT_TAG ?= $(shell git describe --tags --dirty --always)
 
+BASE_REF ?= $(shell git rev-parse --abbrev-ref HEAD)
```
Comment (wzshiming) [inline Makefile:24]: Isn't it a bit much that every commit on a branch builds the image once?
https://github.com/kubernetes/test-infra/pull/27128/files#diff-658f73e8fc0958a94f1af5d908b874ce2331bcad55cec424bd91172c1924775aR12

File: images/cluster/build.sh [hunk 2]
```diff
@@ -14,26 +14,167 @@
...
+  if [[ "${DRY_RUN}" != "true" ]]; then
+    eval "${@}"
+  fi
+}
+
+function main() {
+  local extra_args
+  local suffix
+  if [[ "${#PLATFORMS}" -ne 0 ]]; then
+    export DOCKER_CLI_EXPERIMENTAL=enabled
```
Comment (yibozhuang) [inline images/cluster/build.sh:115]: do we actually need this?
I thought we should be able to use buildkit without needing to set `DOCKER_CLI_EXPERIMENTAL`

Comment (wzshiming) [reply images/cluster/build.sh:115]: `docker buildx` requires this to work.

**Review by yibozhuang — APPROVED** [review]
/lgtm

**Comment by wzshiming** [comment]
@yibozhuang 

Change registry and tag, LGTM again.
https://github.com/kubernetes-sigs/kwok/pull/18/commits/ca6f9c366d686e59c51ccdb87c0e3fd0f72dde3e

**Review by yibozhuang — APPROVED** [review]
/lgtm

**Pull request merged**